composer/distributed/mosaic_fsdp.py

```diff
@@ -0,0 +1,258 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""FSDP related configs and helper functions."""
+
+import logging
+import warnings
+from typing import Any, Dict, Tuple, Union
+
+import torch
+from packaging import version
+from torch import distributed
+from torch.distributed import ProcessGroup
+from torch.distributed.fsdp import (
+    BackwardPrefetch,
+    CPUOffload,
+    MixedPrecision,
+    ShardingStrategy,
+)
+
+from composer.core import Precision
+from composer.utils import VersionedDeprecationWarning, dist
+
+log = logging.getLogger(__name__)
+
+SHARDING_MAP = {
```

Comment:
Enum?

Reply:
Existing code, not going to change given already big PR.